ui/apps/platform/src/services/ComplianceScanConfigurationService.ts

```diff
@@ -84,10 +84,13 @@ export type ComplianceScanConfigurationStatus = {
 // @TODO: This may change and be moved around depending on how backend implements it.
 export type ComplianceReportSnapshot = {
     reportJobId: string;
+    scanConfigId: string;
```

Comment:
Aha, `reportConfigId` and `scanConfigId` will differ.

What do you think about pro and con of as-lightweight-as-possible abstraction object to encapsulate access to the snapshot objects? Report job status needs the following, and anything else?
* job id
* config id
* name
* description
* download function, or maybe higher order function to return the function?

Reply:
Yea, I think thats a good approach. I was thinking we could make a base type called `Snapshot` that is used for both `ComplianceReportSnapshot` and `ReportSnapshot`. So something along these lines:

```
export type Snapshot = {
    reportJobId: string;
    name: string;
    description?: string;
    reportStatus: ReportStatus;
    user: SlimUser;
    isDownloadAvailable: boolean;
};
```
and then each of those types can utilize it:

```
// apps/platform/src/services/ReportsService.types.ts

export type ReportSnapshot = Snapshot & {
    reportConfigId: string;
    vulnReportFilters: VulnerabilityReportFilters;
    collectionSnapshot: CollectionSnapshot;
    schedule: Schedule | null;
    notifiers: NotifierConfiguration[];
};
```

```
// apps/platform/src/services/ComplianceScanConfigurationService.ts

export type ComplianceReportSnapshot = Snapshot & {
    scanConfigId: string;
    reportData: ComplianceScanConfigurationStatus;
};
```

Reply:
Now, there is a question of where this base type should live. @dvail had some suggestions below which I believe are related to this. I think we could keep this type within the global `types` directory. It will be a dependency for both `Components` and `Containers`. Another option is to keep it within the `services` directory, but i'm not sure how that would look.

Reply:
What you did looks good for now.

Because I review for other teams who are willing to make corresponding updates to backend and frontend response type, it is helpful if there is a pattern so they can find a file:
* If in backend proto/api subfolders, then src/services works well
* If in backend proto/storage (but only applies to older endpoints) then src/types works well.